src/core/k8.mjs

```diff
@@ -700,13 +665,24 @@ export class K8 {
   async portForward (podName, localPort, podPort) {
     const ns = this._getNamespace()
     const forwarder = new k8s.PortForward(this.kubeConfig, false)
-    const server = await net.createServer(async (socket) => {
-      K8._webSocketGetters.push(await forwarder.portForward(ns, podName, [podPort], socket, null, socket, 3))
+    const server = await net.createServer((socket) => {
+      forwarder.portForward(ns, podName, [podPort], socket, null, socket, 3)
     })
 
-    K8.makeKillable(server)
+    // add info for logging
+    server.info = `${podName}:${podPort} -> ${constants.LOCAL_HOST}:${localPort}`
+    return server.listen(localPort, constants.LOCAL_HOST)
+  }
 
-    return server.listen(localPort, '127.0.0.1')
+  /**
+   * Stop the port forwarder server
+   *
+   * @param server an instance of server returned by portForward method
+   * @return {Promise<void>}
+   */
+  async stopPortForward (server) {
```

Comment:
I don't think this needs to be async with just calling server.close().  Server.close() is some nasty old cold that uses event management, so when it returns back, it isn't closed.   It will close once it has time to process in the callback chain.

Reply:
Ok, thanks. I'll remove and do some testing.

Reply:
I've removed `async` and made the function return a proper promise that handles the callback of that close function.

Reply:
yes, I dug really deep into that code, because I thought it was the issue causing the node start command to not exit.  it is really meant to be run in a browser, not NodeJS.  I think it used because it works in both and creates code reuse.